addons/hr/models/hr_employee.py

```diff
@@ -187,6 +188,30 @@ def _compute_work_permit_name(self):
             permit_no = '_' + employee.permit_no if employee.permit_no else ''
             employee.work_permit_name = "%swork_permit%s" % (name, permit_no)
 
+    @api.depends('user_id')
+    def _compute_related_partners_count(self):
+        self.related_partners_count = len(self._get_related_partners())
+
+    def _get_related_partners(self):
+        return self.work_contact_id
```

Comment:
```suggestion
        return self.work_contact_id | self.user_id.partner_id
```

Reply:
Hello @Bertrand2 
Changes on the `user_partner_id` will reflect on `work_contact_id`
both values are the same in the end :thinking: